core/src/components/nav/nav.tsx

```diff
@@ -906,25 +889,36 @@ export class Nav implements NavOutlet {
     };
   }
 
+  /**
+   * Inserts a view at the specified index.
+   *
+   * When the view already is in the stack it will be moved to the new position.
+   *
+   * @param view The view to insert.
+   * @param index The index where to insert the view.
+   */
   private insertViewAt(view: ViewController, index: number) {
     const views = this.views;
     const existingIndex = views.indexOf(view);
     if (existingIndex > -1) {
-      // this view is already in the stack!!
-      // move it to its new location
+      // The view already in the stack, removes it.
```

Comment:
It looks like this moves the item in the stack rather than removes it altogether. Can we update the comment?

Reply:
I have clarified the comment.

And I think there is probably a bug here as you probably want to insert at index - 1 if `existingIndex` is before `index`.
I don't want to fix this in this refactoring PR and need to come with a test case.